client/src/main/java/io/split/client/impressions/Impression.java

```diff
@@ -0,0 +1,54 @@
+package io.split.client.impressions;
+
+/**
+ * Created by adilaijaz on 3/23/17.
+ */
+public class Impression {
+    
+    private final String _key;
+    private final String _bucketingKey;
+    private final String _split;
+    private final String _treatment;
+    private final long _time;
+    private final String _appliedRule;
+    private final Long _changeNumber;
+
+
+    public Impression(String key, String bucketingKey, String split, String treatment, long time, String appliedRule, Long changeNumber) {
```

Comment:
we do, but that is an internal API. This is an API meant for external consumption. Reasons for keeping them separate:

- We may be in a world down the road where we are sending more data to ourselves than to an external API
- The internal api says 'label' and 'feature'. For external API, i am using 'appliedRule' and 'spilt' to be more consistent with the UI.